src/package/package.ts

```diff
@@ -37,135 +41,216 @@ const packagePrefixes = {
+  /**
+   * Returns the package ID of the package.
+   *
+   * @returns {string} package ID
+   */
+  public getId(): string {
+    return this.packageId;
   }
 
-  public async getInstallStatus(installRequestId: string): Promise<PackageInstallRequest> {
-    return getStatus(this.options.connection, installRequestId);
+  /**
+   * Returns the package type of the package.
+   *
+   * @returns {Promise<PackageType>}
+   */
+  public async getType(): Promise<PackageType> {
+    return (await this.getPackageData()).ContainerOptions;
   }
 
-  public list(): Promise<QueryResult<PackagingSObjects.Package2>> {
-    return listPackages(this.options.connection);
-  }
+  /**
+   * Returns the list of package versions for the package.
+   * See {@link PackageVersionListOptions} for list options
+   *
+   * @param options
+   * @returns {Promise<PackageVersionListResult[]>}
+   */
+  public async getPackageVersions(options?: PackageVersionListOptions): Promise<PackageVersionListResult[]> {
+    // This should be calling PackageVersion.list() here, but that method is not implemented yet.
+    const packageOptions = {
+      packages: [this.packageId],
+    };
 
-  public async uninstall(
-    id: string,
-    wait: Duration
-  ): Promise<PackagingSObjects.SubscriberPackageVersionUninstallRequest> {
-    return await uninstallPackage(id, this.options.connection, wait);
+    return Package.listVersions(this.options.connection, this.options.project, {
+      ...packageOptions,
+      ...options,
+    } as PackageVersionListOptions);
   }
 
   /**
-   * Reports on the uninstall progress of a package.
+   * Deletes the package.
    *
-   * @param id the 06y package uninstall request id
    */
-  public async uninstallReport(id: string): Promise<PackagingSObjects.SubscriberPackageVersionUninstallRequest> {
-    const result = (await this.options.connection.tooling.retrieve(
-      'SubscriberPackageVersionUninstallRequest',
-      id
-    )) as PackagingSObjects.SubscriberPackageVersionUninstallRequest;
-    if (result.Status === 'Error') {
-      const errorDetails = await this.options.connection.tooling.query<{ Message: string }>(
-        `SELECT Message FROM PackageVersionUninstallRequestError WHERE ParentRequest.Id = '${id}' ORDER BY Message`
-      );
-      const errors: string[] = [];
-      errorDetails.records.forEach((record) => {
-        errors.push(`(${errors.length + 1}) ${record.Message}`);
-      });
-      const errHeader = errors.length > 0 ? `\n=== Errors\n${errors.join('\n')}` : '';
-      const err = messages.getMessage('defaultErrorMessage', [id, result.Id]);
-
-      throw new SfError(`${err}${errHeader}`, 'UNINSTALL_ERROR', [messages.getMessage('action')]);
+  public async delete(): Promise<PackageSaveResult> {
+    const updateResult = await this.options.connection.tooling.delete('Package2', [this.packageId]).catch((err) => {
+      throw applyErrorAction(massageErrorMessage(err as Error));
+    });
+    if (updateResult[0]?.success) {
```

Comment:
should this be `!updateResult[0]?.success`? If it's correct maybe adding a comment why a successful delete throws an error

Reply:
The catch of line 207 handles any fatal errors that occur while processing the delete. The condition checks if the delete operation was successful. This code will be changing on next push.